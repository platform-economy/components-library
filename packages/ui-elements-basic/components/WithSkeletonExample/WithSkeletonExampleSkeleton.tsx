import * as React from 'react';
import { Basic } from '@relayr/ui-elements-skeletons';

export type WithSkeletonExampleSkeletonProps = {};

export class WithSkeletonExampleSkeleton extends React.Component<WithSkeletonExampleSkeletonProps> {

  public render() {
    return (
      <Basic height="20px" width="200px" />
    );
  }
}
