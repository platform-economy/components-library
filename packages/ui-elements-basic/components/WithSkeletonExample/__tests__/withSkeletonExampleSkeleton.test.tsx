import * as React from 'react';
import * as TestRenderer from 'react-test-renderer';

jest.mock('@relayr/ui-elements-skeletons', () => ({
  // tslint:disable-next-line:no-any
  Basic: (props: any) => (
    <div id="BasicSkeleton">{props.children}</div>
  ),
}));

// Component
import { WithSkeletonExampleSkeleton } from '../WithSkeletonExampleSkeleton';

describe('ui-elements-basic', () => {
  describe('Components', () => {
    describe('WithSkeletonExampleSkeleton', () => {

      it('should render with default properties only', () => {
        const tree = TestRenderer
          .create(<WithSkeletonExampleSkeleton />)
          .toJSON();
        expect(tree).toMatchSnapshot();
      });
    });
  });
});
