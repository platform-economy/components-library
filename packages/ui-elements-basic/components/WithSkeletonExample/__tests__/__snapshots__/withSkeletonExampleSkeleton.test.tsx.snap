// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ui-elements-basic Components WithSkeletonExampleSkeleton should render with default properties only 1`] = `
<div
  id="BasicSkeleton"
/>
`;
